classes/blocks/class-socialmedia.php

```diff
@@ -123,7 +131,13 @@ public function prepare_data( $fields ): array {
 					$url = add_query_arg( [ 'omit_script' => true ], $url );
 				}
 				if ( in_array( $provider, self::ALLOWED_OEMBED_PROVIDERS, true ) ) {
-					$data['embed_code'] = wp_oembed_get( $url );
+
+					if ( 'twitter' === $provider ) {
+						$data['embed_code'] = wp_oembed_get( $url );
+					} else {
+						$body               = $this->get_fb_oembed_html( rawurlencode( $url ), $provider );
+						$data['embed_code'] = $body['html'];
```

Comment:
There's no `html` entry when you return en error (it returns just a string, so there's no error message displayed in the end); we could use just `$body`, and return `json_decode( $response['body']['html'], true );` in `get_fb_oembed_html()`